client/jetpack-connect/controller.js

```diff
@@ -111,6 +111,11 @@ export function offerResetRedirects( context, next ) {
 		return externalRedirect( CALYPSO_PLANS_PAGE + selectedSite.slug );
 	}
 
+	// If selected site has a Free plan, redirect to URL passed as query param or wpadmin
+	if ( selectedSite.plan.is_free ) {
+		externalRedirect( context.query.redirect || selectedSite.options.admin_url );
```

Comment:
I would use a `return` statement to avoid the rest of the controller from being executed (the pricing flashes sometimes because of this).

Reply:
Good catch, thanks!